public/js/reports/reports.service.js

```diff
@@ -79,5 +80,33 @@
 
             return columnDescription;
         };
+
+        function checkPrompts (prompts) {
+            const promptsStatus = {};
+            for (const p of prompts) {
+                let isValid;
+                if (p.elementType === 'date') {
+                    if (p.filterType === 'between') {
+                        isValid = !!(p.criterion.date1 && p.criterion.date2);
+                    } else if (p.filterType === 'equal-pattern') {
+                        isValid = (p.criterion.label && p.criterion.datePattern) !== '';
+                    } else {
+                        isValid = !!(p.criterion.date1);
+                    }
+                } else {
+                    if (p.filterType === 'between') {
+                        isValid = 'text1' in p.criterion && 'text2' in p.criterion;
+                    } else {
+                        isValid = 'text1' in p.criterion;
+                    }
+                }
+                promptsStatus[p.id + p.filterType] = isValid;
+            }
+            if (Object.values(promptsStatus).every(e => e === true)) {
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
`Array.prototype.every` already returns a boolean, so the whole if/else block is not needed.

When you have

```js
if (booleanValue) {
    return true;
} else {
    return false;
}
```

It can always be simplified to:

```js
return booleanValue;
```

Reply:
yes you're totally right